doc/md/examples/grammar.txt

```diff
@@ -66,10 +66,19 @@
 <typ_args> ::= 
     '<' <list(<typ>, ',')> '>'
 
+<inst> ::= 
+    <empty>
+    '<' <list(<typ>, ',')> '>'
+    '<' 'system' (',' <typ>)* '>'
+
+<typ_params_opt> ::= 
+    ('<' <list(<typ_bind>, ',')> '>')?
+    '<' 'system' (',' <typ_bind>)* '>'
+
 <typ_field> ::= 
-    'type' <id> ('<' <list(<typ_bind>, ',')> '>')? '=' <typ>
+    'type' <id> <typ_params_opt> '=' <typ>
```

Comment:
I would not use typ_params_opt here since that allows system capabilties on type definitions, which make no sense:

```
[nix-shell:~/motoko/test/fail/ok]$ moc
Motoko compiler (source 0.10.4-82-gfb1a1a4da-dirty)
> type t<system> = Nat;
type t<$> = Nat
> f^C

```
I would actually be simplest just to rule this out syntactically with a different grammar rule for type constructor parameters and arguments!

Reply:
UPDATE:
Perhaps system type arguments to type constructos are already ok (due to the grammar perhaps)?
E.g.
```
type U = List<system> is rejected synactically...
```

Reply:
It is rejected, because it is a different production. (To be checked!)

But look:
```
> class Hey<system>() : {} {setTimer<system>(0, false, func() : async () {});};
stdin:2.27-2.75: type error [M0197], `system` capability required, but not available
 (need an enclosing async expression or function body or explicit `system` type parameter)
```

Reply:
Fixed!